controllers/containerizedworkload_controller.go

```diff
@@ -68,21 +66,29 @@ func (r *ContainerizedWorkloadReconciler) Reconcile(req ctrl.Request) (ctrl.Resu
 
 	var workload oamv1alpha2.ContainerizedWorkload
 	if err := r.Get(ctx, req.NamespacedName, &workload); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Container workload is deleted")
+		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	log.Info("Get the workload", "apiVersion", workload.APIVersion, "kind", workload.Kind)
 
-	deploy, err := r.renderWorkload(ctx, &workload)
+	deploy, err := r.renderDeployment(ctx, &workload)
```

Comment:
You can just plugin your translation function to https://github.com/crossplane/crossplane-runtime/blob/03316b1fbc09b26bcda6cbfc0f62d7a288f3ab43/pkg/reconciler/oam/workload/reconciler.go#L114 and it will take care of Create / Update / Delete for you if you like

Reply:
This just calls the translator (plus some fix). Plus it's good to keep some flexibility here as it takes time to fix things in another repro and then import it.  We probably need to revisit all the repro settings when the converge settles.